base/array.jl

```diff
@@ -66,7 +66,11 @@ strides{T}(a::Array{T,1}) = (1,)
 strides{T}(a::Array{T,2}) = (1, size(a,1))
 strides{T}(a::Array{T,3}) = (1, size(a,1), size(a,1)*size(a,2))
 
-isassigned(a::Array, i::Int...) = isdefined(a, i...)
+function isassigned{T}(a::Array{T}, i::Int...)
+    ii = sub2ind(size(a), i...)
+    1 <= ii <= length(a) || return false
+    ccall(:jl_array_isassigned, Cint, (Any, UInt), a, ii-1) == 1
+end
```

Comment:
This takes a step in the direction of deprecating `isdefined` for Arrays, but you can't add deprecations for builtins. So it stops short.